src/rnp/fficli.cpp

```diff
@@ -1466,20 +1466,27 @@ rnp_cfg_set_defkey(rnp_cfg_t *cfg)
     }
 
     /* If a userid has been given, we'll use it. */
-    if (!(userid = rnp_cfg_getstr(cfg, CFG_USERID))) {
+    userid = NULL;
+    list ids = NULL;
+    if (rnp_cfg_copylist_str(cfg, &ids, CFG_USERID)) {
+        userid = (char*)list_front(ids);
+    }
```

Comment:
Instead of creating temporary list and copying data to it (and duplicate code below), it would be better to add helper function like `const std::string &rnp_cfg_getlist_string(const rnp_cfg_t *cfg, const std::string &key, size_t index)`. Inside you may use `rnp_cfg_getlist()` to check list's length and get value via `rnp_cfg_val_getstr()`.

Reply:
The rnp_cfg_val_getstr() doesn't support the RNP_CFG_VAL_LIST type.

Reply:
@joke325 yeah, I mean something like `rnp_cfg_val_getstr((rnp_cfg_val_t*) list_front(rnp_cfg_getlist(cfg, name)))`